### ISSUE TITLE
Create a Sphinx theme which matches the current website

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7635]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)| @khorn reported|
|-|-|
|Trac ID|trac#7635|
|Type|enhancement|
|Created|2014-09-14 21:36:12Z|
|Branch|https://github.com/twisted/twisted/tree/new-sphinx-theme-7635|

The current Sphinx theme being used to build the narrative docs ('twistedtrac') matches an older version of the website.  We should add a new theme that matches the existing version of the website.

<details><summary>Searchable metadata</summary>

```
trac-id__7635 7635
type__enhancement enhancement
reporter__khorn khorn
priority__normal normal
milestone__ 
branch__branches_new_sphinx_theme_7635 branches/new-sphinx-theme-7635
branch_author__khorn khorn
status__new new
resolution__None None
component__core core
keywords__documentation documentation
time__1410730572623866 1410730572623866
changetime__1416238550162664 1416238550162664
version__None None
owner__khorn khorn

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_1"></a>|@khorn commented|
|-|-|

(In [43068]) Branching to new-sphinx-theme-7635.

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_2"></a>|@khorn set owner to @glyph|
|-|-|

Added a new sphinx theme called "twistednewtrac" in branch `new-sphinx-theme-7635`.

Please advise if you have a better name for the theme, as it's a terrible terrible name.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban commented|
|-|-|

I don't know to much about the process of updating the website... but why create a new theme and why not update the current theme?

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set owner to @khorn|
|-|-|

I'm inclined to agree with Adi.  Can you modify this so it's just an update to the existing theme?  It would be kind of ridiculous to actually maintain 2 different themes that both look like our trac theme.

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_5"></a>|@khorn commented|
|-|-|

I created the new theme in parallel because I thought I might need to build the old one for comparison.  And then I left it because I thought the reviewer might want to do the same.  I had planned to note this on the ticket...but forgot. Sigh.

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_6"></a>|@khorn set owner to @glyph|
|-|-|

OK, I've merged the themes in the branch.  In case it isn't obvious (and it might not be), essentially what's changed is: 

 * some images have been removed, because they are no longer needed
 * the trac banner image has been replaced with the new one
 * the trac.css file has been replace with the current one
 * there are a couple of small tweaks in twistedtrac.css and layout.html

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_7"></a>|@khorn commented|
|-|-|

To be clear, what I've listed above as changes are relative to trunk, not to the previous commit in the branch.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_8"></a>|@glyph removed owner|
|-|-|

Un-assigning all review tickets assigned to me.  Unless there's a specific reason _I_ need to review something, let's leave them unassigned :-).

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_9"></a>|@adiroiban set owner to @khorn|
|-|-|

Thanks for the changes!

1. Do we want to force visitors to HTTPS? Maybe yes, but then we should also change http://tm.com to redirect to https://tm.com ... otherwise, maybe is better to use ```/trac/wiki``` instead of ```https://twistedmatrix.com/trac/wiki```

2. ```+<!--       <p class="left2">``` In case the code is not used, I prefer to just remove it... VC tool will take care of remembering it.


3. ``` +/* Not sure about this one */ ``` please add a better comment or remove it :)
 
4. Documentation has a white background while the main site has a pattern on the background

5. ```/* Modified */``` do we want to keep them ? I see some are removed... but other were kept. I prefer to remove them all.

---------

6. This is just a small architecture comment.. should the documentation has a different CSS file than the general CSS file used by the main site? or just have it applied on top of main website. In this way there is no need for the modified comment.

6.1. Should documenation ship with the images files, why not reuse the ones from the website ?

but maybe this will just make things more complicated as it will be harder to view the documentation offline (outside of the main website)


-------

I have build to documentation and compared the header with the main website, they look the same.


Thanks!
